<?xml version="1.0" encoding="utf-8" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="Docutils 0.11: http://docutils.sourceforge.net/" />
<title>Mutations at site 166 in HA of human pandemic H1N1</title>
<style type="text/css">

/*
:Author: David Goodger (goodger@python.org)
:Id: $Id: html4css1.css 7614 2013-02-21 15:55:51Z milde $
:Copyright: This stylesheet has been placed in the public domain.

Default cascading style sheet for the HTML output of Docutils.

See http://docutils.sf.net/docs/howto/html-stylesheets.html for how to
customize this style sheet.
*/

/* used to remove borders from tables and images */
.borderless, table.borderless td, table.borderless th {
  border: 0 }

table.borderless td, table.borderless th {
  /* Override padding for "table.docutils td" with "! important".
     The right padding separates the table cells. */
  padding: 0 0.5em 0 0 ! important }

.first {
  /* Override more specific margin styles with "! important". */
  margin-top: 0 ! important }

.last, .with-subtitle {
  margin-bottom: 0 ! important }

.hidden {
  display: none }

a.toc-backref {
  text-decoration: none ;
  color: black }

blockquote.epigraph {
  margin: 2em 5em ; }

dl.docutils dd {
  margin-bottom: 0.5em }

object[type="image/svg+xml"], object[type="application/x-shockwave-flash"] {
  overflow: hidden;
}

/* Uncomment (and remove this text!) to get bold-faced definition list terms
dl.docutils dt {
  font-weight: bold }
*/

div.abstract {
  margin: 2em 5em }

div.abstract p.topic-title {
  font-weight: bold ;
  text-align: center }

div.admonition, div.attention, div.caution, div.danger, div.error,
div.hint, div.important, div.note, div.tip, div.warning {
  margin: 2em ;
  border: medium outset ;
  padding: 1em }

div.admonition p.admonition-title, div.hint p.admonition-title,
div.important p.admonition-title, div.note p.admonition-title,
div.tip p.admonition-title {
  font-weight: bold ;
  font-family: sans-serif }

div.attention p.admonition-title, div.caution p.admonition-title,
div.danger p.admonition-title, div.error p.admonition-title,
div.warning p.admonition-title, .code .error {
  color: red ;
  font-weight: bold ;
  font-family: sans-serif }

/* Uncomment (and remove this text!) to get reduced vertical space in
   compound paragraphs.
div.compound .compound-first, div.compound .compound-middle {
  margin-bottom: 0.5em }

div.compound .compound-last, div.compound .compound-middle {
  margin-top: 0.5em }
*/

div.dedication {
  margin: 2em 5em ;
  text-align: center ;
  font-style: italic }

div.dedication p.topic-title {
  font-weight: bold ;
  font-style: normal }

div.figure {
  margin-left: 2em ;
  margin-right: 2em }

div.footer, div.header {
  clear: both;
  font-size: smaller }

div.line-block {
  display: block ;
  margin-top: 1em ;
  margin-bottom: 1em }

div.line-block div.line-block {
  margin-top: 0 ;
  margin-bottom: 0 ;
  margin-left: 1.5em }

div.sidebar {
  margin: 0 0 0.5em 1em ;
  border: medium outset ;
  padding: 1em ;
  background-color: #ffffee ;
  width: 40% ;
  float: right ;
  clear: right }

div.sidebar p.rubric {
  font-family: sans-serif ;
  font-size: medium }

div.system-messages {
  margin: 5em }

div.system-messages h1 {
  color: red }

div.system-message {
  border: medium outset ;
  padding: 1em }

div.system-message p.system-message-title {
  color: red ;
  font-weight: bold }

div.topic {
  margin: 2em }

h1.section-subtitle, h2.section-subtitle, h3.section-subtitle,
h4.section-subtitle, h5.section-subtitle, h6.section-subtitle {
  margin-top: 0.4em }

h1.title {
  text-align: center }

h2.subtitle {
  text-align: center }

hr.docutils {
  width: 75% }

img.align-left, .figure.align-left, object.align-left {
  clear: left ;
  float: left ;
  margin-right: 1em }

img.align-right, .figure.align-right, object.align-right {
  clear: right ;
  float: right ;
  margin-left: 1em }

img.align-center, .figure.align-center, object.align-center {
  display: block;
  margin-left: auto;
  margin-right: auto;
}

.align-left {
  text-align: left }

.align-center {
  clear: both ;
  text-align: center }

.align-right {
  text-align: right }

/* reset inner alignment in figures */
div.align-right {
  text-align: inherit }

/* div.align-center * { */
/*   text-align: left } */

ol.simple, ul.simple {
  margin-bottom: 1em }

ol.arabic {
  list-style: decimal }

ol.loweralpha {
  list-style: lower-alpha }

ol.upperalpha {
  list-style: upper-alpha }

ol.lowerroman {
  list-style: lower-roman }

ol.upperroman {
  list-style: upper-roman }

p.attribution {
  text-align: right ;
  margin-left: 50% }

p.caption {
  font-style: italic }

p.credits {
  font-style: italic ;
  font-size: smaller }

p.label {
  white-space: nowrap }

p.rubric {
  font-weight: bold ;
  font-size: larger ;
  color: maroon ;
  text-align: center }

p.sidebar-title {
  font-family: sans-serif ;
  font-weight: bold ;
  font-size: larger }

p.sidebar-subtitle {
  font-family: sans-serif ;
  font-weight: bold }

p.topic-title {
  font-weight: bold }

pre.address {
  margin-bottom: 0 ;
  margin-top: 0 ;
  font: inherit }

pre.literal-block, pre.doctest-block, pre.math, pre.code {
  margin-left: 2em ;
  margin-right: 2em }

pre.code .ln { color: grey; } /* line numbers */
pre.code, code { background-color: #eeeeee }
pre.code .comment, code .comment { color: #5C6576 }
pre.code .keyword, code .keyword { color: #3B0D06; font-weight: bold }
pre.code .literal.string, code .literal.string { color: #0C5404 }
pre.code .name.builtin, code .name.builtin { color: #352B84 }
pre.code .deleted, code .deleted { background-color: #DEB0A1}
pre.code .inserted, code .inserted { background-color: #A3D289}

span.classifier {
  font-family: sans-serif ;
  font-style: oblique }

span.classifier-delimiter {
  font-family: sans-serif ;
  font-weight: bold }

span.interpreted {
  font-family: sans-serif }

span.option {
  white-space: nowrap }

span.pre {
  white-space: pre }

span.problematic {
  color: red }

span.section-subtitle {
  /* font-size relative to parent (h1..h6 element) */
  font-size: 80% }

table.citation {
  border-left: solid 1px gray;
  margin-left: 1px }

table.docinfo {
  margin: 2em 4em }

table.docutils {
  margin-top: 0.5em ;
  margin-bottom: 0.5em }

table.footnote {
  border-left: solid 1px black;
  margin-left: 1px }

table.docutils td, table.docutils th,
table.docinfo td, table.docinfo th {
  padding-left: 0.5em ;
  padding-right: 0.5em ;
  vertical-align: top }

table.docutils th.field-name, table.docinfo th.docinfo-name {
  font-weight: bold ;
  text-align: left ;
  white-space: nowrap ;
  padding-left: 0 }

/* "booktabs" style (no vertical lines) */
table.docutils.booktabs {
  border: 0px;
  border-top: 2px solid;
  border-bottom: 2px solid;
  border-collapse: collapse;
}
table.docutils.booktabs * {
  border: 0px;
}
table.docutils.booktabs th {
  border-bottom: thin solid;
  text-align: left;
}

h1 tt.docutils, h2 tt.docutils, h3 tt.docutils,
h4 tt.docutils, h5 tt.docutils, h6 tt.docutils {
  font-size: 100% }

ul.auto-toc {
  list-style-type: none }

</style>
</head>
<body>
<div class="document" id="mutations-at-site-166-in-ha-of-human-pandemic-h1n1">
<h1 class="title">Mutations at site 166 in HA of human pandemic H1N1</h1>

<p>Analysis of mutations at site 166 (H3 numbering, see <a class="reference internal" href="#ha-numbering">HA numbering</a> below) of the HA from human pandemic H1N1.</p>
<p>The input files, analysis scripts, and results can be downloaded <a class="reference external" href="https://github.com/jbloom/pdmH1N1_HA_K166_mutations">on GitHub</a>.</p>
<p>This analysis was performed by <a class="reference external" href="http://research.fhcrc.org/bloom/en.html">Jesse Bloom</a>.</p>
<div class="contents topic" id="contents">
<p class="topic-title first">Contents</p>
<ul class="simple">
<li><a class="reference internal" href="#methods-summary" id="id5">Methods summary</a></li>
<li><a class="reference internal" href="#ha-numbering" id="id6">HA numbering</a></li>
<li><a class="reference internal" href="#software-used" id="id7">Software used</a></li>
<li><a class="reference internal" href="#input-files" id="id8">Input files</a></li>
<li><a class="reference internal" href="#analysis" id="id9">Analysis</a><ul>
<li><a class="reference internal" href="#sequence-alignment-sequence-selection-mutation-counts" id="id10">Sequence alignment, sequence selection, mutation counts</a></li>
<li><a class="reference internal" href="#plotting-amino-acid-fractions" id="id11">Plotting amino-acid fractions</a></li>
<li><a class="reference internal" href="#construction-of-phylogenetic-tree" id="id12">Construction of phylogenetic tree</a></li>
</ul>
</li>
</ul>
</div>
<div class="section" id="methods-summary">
<h1><a class="toc-backref" href="#id5">Methods summary</a></h1>
<p>The occurrence of different amino-acid identities at HA residue 166 (H3 numbering) was analyzed by downloading all full-length human pandemic H1N1 sequences present in the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/genomes/FLU/FLU.html">Influenza Virus Resource</a> <a class="citation-reference" href="#bao2008" id="id1">[Bao2008]</a> as of Feb-23-2014. After purging sequences that were less than full length, contained ambiguous nucleotide identities, lacked full (year, month, day) isolation dates, or were otherwise anomalous, the sequences were aligned. Each calendar year was broken into four equal partitions beginning with January 1, the frequencies of different amino acids at residue 166 for each partition was calculated and plotted. For construction of phylogenetic trees, the sequence set was randomly subsampled to 10 sequences per quarter-year partition.
<a class="reference external" href="http://beast.bio.ed.ac.uk/Main_Page">BEAST</a> <a class="citation-reference" href="#drummond2013" id="id2">[Drummond2013]</a> was then used to sample from the posterior distribution of phylogenetic trees with reconstructed sequences at the nodes, after date stamping the sequences, using a <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/1633570">JTT</a> <a class="citation-reference" href="#jones1992" id="id3">[Jones1992]</a> with a single rate category with an exponential prior, a strict molecular clock, and relatively uninformative coalescent-based prior over the tree. The figure shows a maximum clade credibility summary of the posterior distribution with branches colored according to the reconstructed amino-acid identity with the highest posterior probability at their descendent nodes. The tree was visually rendered using <a class="reference external" href="http://tree.bio.ed.ac.uk/software/figtree/">FigTree</a>. The input data and computer code used for this analysis can be found <a class="reference external" href="https://github.com/jbloom/pdmH1N1_HA_K166_mutations">on GitHub</a> at <a class="reference external" href="https://github.com/jbloom/pdmH1N1_HA_K166_mutations">https://github.com/jbloom/pdmH1N1_HA_K166_mutations</a>.</p>
</div>
<div class="section" id="ha-numbering">
<h1><a class="toc-backref" href="#id6">HA numbering</a></h1>
<p>Various numbering schemes are in use for influenza HA (see <a class="reference external" href="https://github.com/jbloom/HA_numbering">HA_numbering</a>). The site examined here is:</p>
<ul class="simple">
<li>Residue 166 in the H3 numbering scheme (for example, as in PDB <a class="reference external" href="http://www.rcsb.org/pdb/explore.do?structureId=4HMG">4HMG</a>).</li>
<li>Residue 169 in the H1 numbering scheme (for example, as in PDB <a class="reference external" href="http://www.rcsb.org/pdb/explore.do?structureId=4JTV">4JTV</a>).</li>
<li>Residue 180 in sequential 1, 2, ... numbering of typical pandemic H1N1 HAs beginning with 1 a the N-terminal methionine.</li>
</ul>
</div>
<div class="section" id="software-used">
<h1><a class="toc-backref" href="#id7">Software used</a></h1>
<ul class="simple">
<li><a class="reference external" href="http://www.python.org/">Python</a> : version 2.7.5</li>
<li><a class="reference external" href="http://emboss.sourceforge.net/apps/release/6.6/emboss/apps/needle.html">EMBOSS needle</a> : from EMBOSS 6.3.1, for sequence alignments.</li>
<li><a class="reference external" href="http://jbloom.github.io/mapmuts/">mapmuts</a>: version 1.0, for sequence manipulation utilities.</li>
<li><a class="reference external" href="http://matplotlib.org/">matplotlib</a> : version 1.3.1, for plotting.</li>
<li><a class="reference external" href="http://www.imagemagick.org/script/index.php">ImageMagick convert</a> : version 6.8.3-3, for converting PDF images to JPGs.</li>
<li><a class="reference external" href="http://beast.bio.ed.ac.uk/Main_Page">BEAST</a> : version 1.8.0, for inferring phylogenetic tree. Executables used from the package are <tt class="docutils literal">beast</tt> and <tt class="docutils literal">treeannotator</tt>.</li>
<li><a class="reference external" href="http://beast.bio.ed.ac.uk/BEAGLE">BEAGLE</a> : version 2.1.2, for use with <a class="reference external" href="http://beast.bio.ed.ac.uk/Main_Page">BEAST</a>.</li>
<li><a class="reference external" href="http://beast.bio.ed.ac.uk/Tracer">Tracer</a> : version 1.6, for analyzing <a class="reference external" href="http://beast.bio.ed.ac.uk/Main_Page">BEAST</a> output.</li>
<li><a class="reference external" href="http://tree.bio.ed.ac.uk/software/figtree/">FigTree</a> : version 1.4, for creating the tree image.</li>
</ul>
</div>
<div class="section" id="input-files">
<h1><a class="toc-backref" href="#id8">Input files</a></h1>
<p>Here are the input sequence data files. All other files are analysis scripts or output files that form part of the <a class="reference internal" href="#analysis">Analysis</a> described below. Here are the input files:</p>
<ul class="simple">
<li><em>HAseqs.fasta</em> : File containing all full-length protein-coding human pandemic H1N1 HA sequences as downloaded from the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/genomes/FLU/FLU.html">Influenza Virus Resource</a> on February-23-2014.</li>
<li><em>California2009-HA.fasta</em> : File containing protein-coding HA from <em>A/California/4/2009</em>, which is used as the reference sequence for alignments.</li>
<li><em>AnomalousSequences.txt</em> : File listing sequences that appear to be anomalous due to much higher than expected divergence. These are probably sequences that were incorrectly classified as human pandemic H1N1 in the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/genomes/FLU/FLU.html">Influenza Virus Resource</a>. These sequences are excluded from the analyses.</li>
</ul>
</div>
<div class="section" id="analysis">
<h1><a class="toc-backref" href="#id9">Analysis</a></h1>
<p>Here are the steps in the analysis:</p>
<div class="section" id="sequence-alignment-sequence-selection-mutation-counts">
<h2><a class="toc-backref" href="#id10">Sequence alignment, sequence selection, mutation counts</a></h2>
<p>The first step is performed by the <a class="reference external" href="http://www.python.org/">Python</a> script <em>parseseqs.py</em>, and can be run with:</p>
<pre class="literal-block">
python parseseqs.py
</pre>
<p>The steps are:</p>
<ol class="arabic">
<li><p class="first">Align the sequences. All sequences in <em>HAseqs.fasta</em> are pairwise aligned with the reference sequence in <em>California2009-HA.fasta</em> after:</p>
<blockquote>
<ol class="loweralpha simple">
<li>Purging any sequences with ambiguous nucleotides, and any sequences that are specified as anomalies in <em>AnomalousSequences.txt</em>.</li>
<li>Purging any sequences that do not exactly match the reference sequence in length.</li>
<li>Purging any sequences that do not have a full date specified (year, month, and day).</li>
</ol>
</blockquote>
</li>
</ol>
<blockquote>
The resulting aligned and translated protein sequences are written to the created file <em>all_alignedproteins.fasta</em>.</blockquote>
<ol class="arabic" start="2">
<li><p class="first">Get a subset of sequences for building the phylogenetic tree. Each year is broken into four partitions spanning months 1-3, 4-6, 7-9, and 10-12. For each partition, up to 10 sequences per year are randomly selected to be retained. These sequences are written to <em>selected_alignedproteins.fasta</em>. The reason for only keeping a subset of the sequence in this way is to have a sequence set small enough to build a phylogenetic tree without undue computational burden.</p>
</li>
<li><p class="first">Count the occurrences of different amino-acid identities at site 166 (H3 numbering, see <a class="reference internal" href="#ha-numbering">HA numbering</a>) for each of the date partitions. The results are written to the file <em>partitioncounts.txt</em>. Here is the raw data in that file:</p>
<blockquote>
<pre class="literal-block">
#Counts of different amino acids in each date partiton.
#date   start   end     nseqs   A       C       D       E       F       G       H       I       K       L       M       N       P       Q       R       S       T       V       W       Y
2009.12 2009.00 2009.25 8       0       0       0       0       0       0       0       0       8       0       0       0       0       0       0       0       0       0       0       0
2009.38 2009.25 2009.50 1707    0       0       0       1       0       0       0       0       1706    0       0       0       0       0       0       0       0       0       0       0
2009.62 2009.50 2009.75 1929    0       0       0       4       0       0       0       0       1925    0       0       0       0       0       0       0       0       0       0       0
2009.88 2009.75 2010.00 1885    0       0       0       0       0       0       0       2       1872    0       0       10      0       0       1       0       0       0       0       0
2010.12 2010.00 2010.25 399     0       0       0       0       0       0       0       0       395     0       0       2       0       2       0       0       0       0       0       0
2010.38 2010.25 2010.50 125     0       0       0       0       0       0       0       0       124     0       0       0       0       0       0       0       1       0       0       0
2010.62 2010.50 2010.75 241     0       0       0       1       0       0       0       0       233     0       0       0       0       1       0       0       6       0       0       0
2010.88 2010.75 2011.00 209     0       0       0       0       0       0       0       1       205     0       0       3       0       0       0       0       0       0       0       0
2011.12 2011.00 2011.25 437     0       0       0       0       0       0       0       2       410     0       0       2       0       1       3       1       18      0       0       0
2011.38 2011.25 2011.50 48      0       0       0       0       0       0       0       0       44      0       0       0       0       0       0       0       4       0       0       0
2011.62 2011.50 2011.75 26      0       0       0       0       0       0       0       0       26      0       0       0       0       0       0       0       0       0       0       0
2011.88 2011.75 2012.00 17      0       0       0       0       0       0       0       0       17      0       0       0       0       0       0       0       0       0       0       0
2012.12 2012.00 2012.25 171     0       0       0       1       0       0       0       0       169     0       0       0       0       0       1       0       0       0       0       0
2012.38 2012.25 2012.50 26      0       0       0       2       0       0       0       0       22      0       0       2       0       0       0       0       0       0       0       0
2012.62 2012.50 2012.75 16      0       0       0       0       0       0       0       1       15      0       0       0       0       0       0       0       0       0       0       0
2012.88 2012.75 2013.00 47      0       0       0       0       0       0       0       6       34      0       0       0       0       7       0       0       0       0       0       0
2013.12 2013.00 2013.25 327     0       0       0       0       0       0       0       41      253     0       0       4       0       27      0       0       2       0       0       0
2013.38 2013.25 2013.50 36      0       0       0       4       0       0       0       1       9       0       0       1       0       21      0       0       0       0       0       0
2013.62 2013.50 2013.75 17      0       0       0       1       0       0       0       0       0       0       0       1       0       15      0       0       0       0       0       0
2013.88 2013.75 2014.00 43      0       0       0       0       0       0       0       0       0       0       0       0       0       43      0       0       0       0       0       0

</pre>
</blockquote>
</li>
</ol>
</div>
<div class="section" id="plotting-amino-acid-fractions">
<h2><a class="toc-backref" href="#id11">Plotting amino-acid fractions</a></h2>
<p>The <a class="reference external" href="http://www.python.org/">Python</a> script <em>makeplot.py</em> is used to make a PDF plot (with <a class="reference external" href="http://matplotlib.org/">matplotlib</a>) showing the frequencies of different amino acids at site 166 as a function of time. Run the script with:</p>
<pre class="literal-block">
python makeplot.py
</pre>
<p>The resulting plot divides each year into four partitions (months 1-3, 4-6, 7-9, 10-12) as in <em>partitioncounts.txt</em>. The frequencies of different amino acids at site 166 is then shown for each of these partitions.</p>
<p>The created plot is <em>aafracs.pdf</em>. A JPG version (lower quality, created with <a class="reference external" href="http://www.imagemagick.org/script/index.php">ImageMagick convert</a> from the PDF) is also created as <em>aafracs.jpg</em>. Here is that plot:</p>
<blockquote>
<div class="figure align-center">
<img alt="aafracs.jpg" src="aafracs.jpg" style="width: 40%;" />
<p class="caption">The plot <em>aafracs.jpg</em> showing the frequencies of different amino acids at site 166. A higher quality image of this same plot is in <em>aafracs.pdf</em>.</p>
</div>
</blockquote>
<p>This plot is a graphical display of the data in <em>partitioncounts.txt</em>.</p>
</div>
<div class="section" id="construction-of-phylogenetic-tree">
<h2><a class="toc-backref" href="#id12">Construction of phylogenetic tree</a></h2>
<p>A phylogenetic tree was constructed using the subset of sequences in <em>selected_alignedproteins.fasta</em>. This subset only contains 10 sequences per year partition (4 partitions per year) -- the reason for only using some sequences is to make the phylogenetic tree construction tractable.</p>
<p>First, the <tt class="docutils literal">beauti</tt> program of the <a class="reference external" href="http://beast.bio.ed.ac.uk/Main_Page">BEAST</a> package was used to construct an input XML file from <em>selected_alignedproteins.fasta</em>. The tips were stamped with their isolation dates, a <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/1633570">JTT</a> substitution model with a single rate category, and a strict molecular clock was assumed. The tree prior was a relatively uninformative coalescent-based prior and an exponential prior was used over the rate. Ancestral states were reconstructed at all nodes. The MCMC was run for 10 million steps. Full details are available in the created XML file <em>selected_alignedproteins.xml</em>.</p>
<p><a class="reference external" href="http://beast.bio.ed.ac.uk/Main_Page">BEAST</a> was then run using <a class="reference external" href="http://beast.bio.ed.ac.uk/BEAGLE">BEAGLE</a> with the command:</p>
<pre class="literal-block">
~/BEASTv1.8.0/bin/beast -overwrite -beagle_SSE -seed 1 -threads 3 selected_alignedproteins.xml

java -classpath ~/BEASTv1.8.0/lib/beast.jar dr.app.beast.BeastMain -threads 6 -overwrite -beagle_SSE selected_alignedproteins.xml
</pre>
<p>This created the following files:</p>
<pre class="literal-block">
selected_alignedproteins.log
selected_alignedproteins.trees
</pre>
<p>The key data is in the latter file (<em>selected_alignedproteins.trees</em>); however, this file is very large and so is not included in this repository on <a class="reference external" href="https://github.com/jbloom/pdmH1N1_HA_K166_mutations">on GitHub</a> (it can be regenerated using the commands above).</p>
<p><a class="reference external" href="http://beast.bio.ed.ac.uk/Tracer">Tracer</a> was used to analyze the <em>selected_alignedproteins.log</em> file to check for MCMC convergence. If the first 10% of steps (first 1 million steps) are excluded, the remaining 90% (9 million steps) appear to be well converged, and have good effective sample sizes. These results indicate that the MCMC was probably run for a sufficiently large number of steps.</p>
<p>An annotated tree was then constructed using <tt class="docutils literal">logcombiner</tt> and <tt class="docutils literal">treeannotator</tt> from the <a class="reference external" href="http://beast.bio.ed.ac.uk/Main_Page">BEAST</a> package. This tree is created automatically by the <a class="reference external" href="http://www.python.org/">Python</a> script <em>analyze_tree.py</em> with the command:</p>
<pre class="literal-block">
python analyze_tree.py
</pre>
<p>This script creates several output files not included in the package <a class="reference external" href="https://github.com/jbloom/pdmH1N1_HA_K166_mutations">on GitHub</a> (again, they can be regenerated) -- the script also contains hardcoded paths to <a class="reference external" href="http://beast.bio.ed.ac.uk/Main_Page">BEAST</a> executables that you may need to modify for your computer. The result of this script that is retained is the file <em>annotated_maxcredtrees.trees</em>.</p>
<p>The <em>annotated_maxcredtrees.trees</em> file was then opened with <a class="reference external" href="http://tree.bio.ed.ac.uk/software/figtree/">FigTree</a> and manually re-formatted for appealing visual appearance. The formatted image was saved as <em>annotated_maxcredtree.pdf</em>. A JPG version of this file (<em>annotated_maxcredtree.jpg</em>) was then created with <a class="reference external" href="http://www.imagemagick.org/script/index.php">ImageMagick convert</a> using:</p>
<pre class="literal-block">
convert -density 300 annotated_maxcredtree.pdf annotated_maxcredtree.jpg
</pre>
<p>Here is that plot:</p>
<blockquote>
<div class="figure align-center">
<img alt="annotated_maxcredtree.jpg" src="annotated_maxcredtree.jpg" style="width: 80%;" />
<p class="caption">The plot <em>annotated_maxcredtree.jpg</em> showing the frequencies of different amino acids at site 166. A higher quality image of this same plot is in <em>annotated_maxcredtree.pdf</em>. In this plot, the residue is labeled as 180 since <a class="reference external" href="http://beast.bio.ed.ac.uk/Main_Page">BEAST</a> uses consecutive numbering rather than the H3 numbering scheme.</p>
</div>
</blockquote>
<table class="docutils citation" frame="void" id="drummond2013" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id2">[Drummond2013]</a></td><td>Drummond AJ, Suchard MA, Xie D, and Rambaut A. Bayesian phylogenetics with BEAUti and BEAST 1.7. Mol Biol Evol. 29:1969-1973 (2012)</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="bao2008" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id1">[Bao2008]</a></td><td>Bao Y, Dernovoy D, Kiryutin B, Zaslavsky L, Tatusova T, Ostell J, and Lipman D. The influenza virus resource at the National Centery for Biotechnology Information. J Virol. 82:596-601 (2008).</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="jones1992" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id3">[Jones1992]</a></td><td>Jones DT, Taylor WR, and Thornton JM. The rapid generation of mutation data matrices from protein sequences. Comput Appl Biosci. 8:275-282 (1992)</td></tr>
</tbody>
</table>
</div>
</div>
</div>
</body>
</html>
